tests/integration/acp/relationship/add_doc_actor_test/add_doc_actor_with_manager_test.go

```diff
@@ -0,0 +1,1286 @@
+
+                          reader:
+                            types:
+                              - actor
+
+                          writer:
+                            types:
+                              - actor
+
+                          admin:
+                            manages:
+                              - writer
+                            types:
+                              - actor
+
+                          dummy:
+                            types:
+                              - actor
+                `,
+
+				ExpectedPolicyID: expectedPolicyID,
+			},
+
+			testUtils.SchemaUpdate{
+				Schema: fmt.Sprintf(`
+						type Users @policy(
+							id: "%s",
+							resource: "users"
+						) {
+							name: String
+							age: Int
+						}
+					`,
+					expectedPolicyID,
+				),
+			},
+
+			testUtils.CreateDoc{
+				Identity: immutable.Some(1),
+
+				CollectionID: 0,
+
+				Doc: `
+					{
+						"name": "Shahzad",
+						"age": 28
+					}
+				`,
+			},
+
+			testUtils.Request{
+				Identity: immutable.Some(3), // This identity can not read yet.
+
+				Request: `
+					query {
+						Users {
+							_docID
+							name
+							age
+						}
+					}
+				`,
+
+				Results: map[string]any{
+					"Users": []map[string]any{}, // Can't see the documents yet
+				},
+			},
+
+			testUtils.AddDocActorRelationship{ // Make admin / manager
+				RequestorIdentity: 1,
+
+				TargetIdentity: 2,
+
+				CollectionID: 0,
+
+				DocID: 0,
+
+				Relation: "admin",
+
+				ExpectedExistence: false,
+			},
+
+			testUtils.AddDocActorRelationship{ // Admin makes another actor a writer
+				RequestorIdentity: 2,
+
+				TargetIdentity: 3,
+
+				CollectionID: 0,
+
+				DocID: 0,
+
+				Relation: "writer",
+
+				ExpectedExistence: false,
+			},
+
+			testUtils.UpdateDoc{
+				CollectionID: 0,
+
+				Identity: immutable.Some(3), // The other actor can update
+
+				DocID: 0,
+
+				Doc: `
+					{
+						"name": "Shahzad Lone"
+					}
+				`,
+			},
+
+			testUtils.Request{
+				Identity: immutable.Some(3), // The other actor can read
+
+				Request: `
+					query {
+						Users {
+							_docID
+							name
+							age
+						}
+					}
+				`,
+
+				Results: map[string]any{
+					"Users": []map[string]any{
+						{
+							"_docID": "bae-9d443d0c-52f6-568b-8f74-e8ff0825697b",
+							"name":   "Shahzad Lone", // Updated name
+							"age":    int64(28),
+						},
+					},
+				},
+			},
+
+			testUtils.DeleteDoc{
+				CollectionID: 0,
+
+				Identity: immutable.Some(3), // The other actor can delete
+
+				DocID: 0,
+			},
+
+			testUtils.Request{
+				Identity: immutable.Some(3),
+
+				Request: `
+					query {
+						Users {
+							_docID
+							name
+							age
+						}
+					}
+				`,
+
+				Results: map[string]any{ // Check actually deleted
+					"Users": []map[string]any{},
+				},
+			},
+		},
+	}
+
+	testUtils.ExecuteTestCase(t, test)
+}
+
+func TestACP_OwnerMakesAManagerThatGivesItSelfReadAccess_ManagerCanRead(t *testing.T) {
```

Comment:
question: why not to make manager automatically have this permissions? It seems to be pointless to do it separately.

Reply:
This is a sourcehub level decision not a defradb one. I am guessing it was for more fine grain metrics or flexibility in future.

Reply:
the questions is then why are we testing sourcehub functionality so vigorously on defra side? After all, this is just one of potential remote ACP implementations. What if other ACPs don't have such behavior? Or what if it changes on the sourcehub side?

Reply:
> the questions is then why are we testing sourcehub functionality so vigorously on defra side?
I don't consider this as `vigorous` at all. It's just that the manager functionality existed before since acp was implemented but it doesn't become functional or even testable until we have relationship feature. So now that we have the relationship sharing implemented this is important to cover in my opinion as it's something users can now use.

> After all, this is just one of potential remote ACP implementations.
Not sure I understand, but every test runs against both `local` and `sourcehub` implementation. 

> What if other ACPs don't have such behavior?
What is this other ACP you are referring to? Not sure I follow.

> Or what if it changes on the sourcehub side?
Then that is a change that DefraDB must adjust with like with all other third party dependency changes (however I doubt this behavior will change).

Reply:
>  why are we testing sourcehub functionality so vigorously on defra side?

I think it is worth having a proper team conversation about this - the ACP interface is kind of a black-box to Defra, and it should be up to the implementations to ensure they properly satisfy the interface.  Defra just needs to worry about calling correctly.

The maintenance cost for all these tests is not free.

Reply:
> > why are we testing sourcehub functionality so vigorously on defra side?
> 
> I think it is worth having a proper team conversation about this - the ACP interface is kind of a black-box to Defra, and it should be up to the implementations to ensure they properly satisfy the interface. Defra just needs to worry about calling it correctly.
> 
> The maintenance cost for all these tests is not free.

I agree, I do kind of look at this situation as along the same line as `Unit test` vs `Integration test` specially because the "sourcehub unit" at this current stage lacks the level of testing I would feel comfortable with specially when using it to deliver some key privacy/security feature of a database.

Reply:
> the "sourcehub unit" at this current stage lacks the level of testing

Yes, this is a problem, but I very strongly believe that any Source member should be free to correct that instead of over-compensating in what could be seen as the wrong repo.

Reply:
> Source member should be free to correct that instead of feeling like they need to over-compensate in what could be seen as the wrong repo.

This is actually a very good idea, this way the testing can belong in more appropriate areas. For example the DPI stuff `write` and `read` specific permission testing can happy on defra and the general invalid and edge cases can belong on sourcehub.

Reply:
I know it feels more reassuring to have many tests covered on Defra side, but as Andy pointed out, it comes at a cost. Defra should test only defra-related edge cases and some general happy-paths to make sure modules work well together.

Reply:
> What is this other ACP you are referring to? Not sure I follow.

We are thinking for example about implementing UCAN-based remote ACP which relies purely on decentralized permission resolution between peers, i.e. without some entity (like sourcehub) in the middle.

Reply:
I am happy to follow consensus, can point to individual tests that are seemingly out of defra-scope and I will push back only if I do deem it "defra" specific, otherwise will try to move them to sourcehub :)

Reply:
Tracking here: https://github.com/sourcenetwork/defradb/issues/2980